inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/DynamicKafkaDeserializationSchema.java

```diff
@@ -131,12 +140,16 @@ public void deserialize(ConsumerRecord<byte[], byte[]> record, Collector<RowData
         if (keyDeserialization != null) {
             keyDeserialization.deserialize(record.key(), keyCollector);
         }
-
         // project output while emitting values
         outputCollector.inputRecord = record;
         outputCollector.physicalKeyRows = keyCollector.buffer;
-        outputCollector.outputCollector =
-                sourceMetricData == null ? collector : new MetricsCollector<>(collector, sourceMetricData);
+        if (sourceMetricData != null) {
+            MetricsCollector<RowData> metricsCollector = new MetricsCollector<>(collector, sourceMetricData);
+            metricsCollector.resetTimestamp((Long) outputCollector.metadataConverters[consumeTimeIndex].read(record));
```

Comment:
please consider the situation where you cannot get a consumeTimeIndex

Reply:
fix it